spec/features/create_virtual_object_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Use Argo to create a virtual object with constituent objects', type: :feature do
+  let(:start_url) { Settings.argo_url }
+  let(:filename_group) { ['example.tiff', 'example.jp2'] }
+  let(:csv_path) { File.join(DownloadHelpers::PATH, 'virtual-object.csv') }
+  let(:virtual_objects_description) { RandomWord.phrases.next }
+  let(:num_constituents) { 11 }
+
+  before do
+    authenticate!(start_url: start_url, expected_text: 'Welcome to Argo!')
+  end
+
+  scenario do
+    ensure_token
+
+    # Create constituent objects
+    object_druids = []
+
+    num_constituents.times do
+      object_druid = deposit_object(filenames: filename_group)
+      object_druids << object_druid
+    end
+
+    puts object_druids
+
+    # Create parent object
+    parent_druid = deposit_object
+
+    # Avoid occasional errors with preservation workflow
```

Comment:
Can you be more specific about the errors? We don't want to have sleep anywhere, so if you think we need it, at least document what error you are avoiding.

Reply:
I tried taking the sleep out just to see if I was being overly cautious and I got one of the errors again just now: 
`Error: sdr-ingest-transfer : problem with sdr-ingest-transfer (BackgroundJob: 233047): {:errors=>[{:title=>"Preservation error", :detail=>"Version mismatch in /dor/workspace/zh/212/bg/7530/zh212bg7530/metadata/versionMetadata.xml, expected 1, found 2"}]} `

I've also seen an error that says, `Error: validate-bag : Bag validation failure(s) for druid:yj420my5239: [{:required_file_not_found=>"Deposit bag required file /services-disk-stage/store2/deposit/yj420my5239/data/metadata/versionMetadata.xml not found"}] `

I'll test some more with sleep back in, and update the comment to describe the error as an "validate-bag error or sdr-ingest-transfer preservation error with versionMetadata.xml", if that sounds informative enough. If there's wording that would better describe this, let me know and I can put that in.

Reply:
I'm not sure how the sleep would prevent such an error. That error took place asynchronously in the accessionWF/preservationWF robots.

Reply:
@mjgiarlo is it possible that Laura has found a bug in the `Create virtual object` where it's not checking to see if all the constituents are accessioned before messing with them?

Reply:
@jcoyne @lwrubel we have been seeing version-related errors when I deploy the `cocinize-reset-content#3331` branch of DSA, which is currently in QA (not stage). And I am not sure I've figured out how to deal with that bug yet. Were the above errors in QA? (And I assume we've not seen them in stage?)

Reply:
@jcoyne when validating constituents, a number of checks are made, including whether each constituent is either Opened or Accessioned: https://github.com/sul-dlss/dor-services-app/pull/3379/files#diff-65455a7c13e2140bbb2d58b75073ad8d010e5da90bca66b52483f30e1e492507R29-R33 so we are making this check

Reply:
Are the versionMetadata errors on the virtual object side, the constituent side, or both?

Reply:
I've been seeing them on stage, after the constituent and parent objects have been created and the bulk action run. When the virtual object is being created (a v2 being accessioned), the error happens. Here's an example: https://argo-stage.stanford.edu/view/druid:zh212bg7530

Reply:
@lwrubel Oh, interesting. I'm curious if you see them in QA too or if that runs clean.

Reply:
@mjgiarlo: just got it on QA: https://argo-qa.stanford.edu/view/druid:jn028rp2502

Reply:
Also am seeing this error on QA, "Creating some or all virtual objects failed because some objects are not combinable: Problematic objects for druid:mx887np6058: undefined method `externalIdentifier' for #<Event:0x0000000006ec2510>".

Reply:
@lwrubel Well, that's odd. It looks like an event object is being included as a constituent!

Reply:
This was due to my DSA branch assuming VersionService.close returned a Cocina object. It returns an event instead.

Reply:
Yay, we caught a bug! 🐛 🐞 🥾